network/server/tchannelthrift/types.go

```diff
@@ -20,10 +20,19 @@
 
 package tchannelthrift
 
-import "github.com/m3db/m3db/serialize"
+import (
+	"github.com/m3db/m3db/serialize"
+	"github.com/m3db/m3x/instrument"
+)
 
 // Options controls server behavior
 type Options interface {
+	// SetInstrumentOptions sets the instrumentation options
+	SetInstrumentOptions(value instrument.Options) Options
+
+	// InstrumentOptions returns the instrumentation options
+	InstrumentOptions() instrument.Options
```

Comment:
mind changing the `NewService()` ctor to use this instead of `s.db.Options().InstrumentOptions()`

Reply:
Sure thing.